presto-main/src/main/java/com/facebook/presto/cost/StatsCalculatorModule.java

```diff
@@ -56,4 +57,23 @@ public static StatsCalculator createNewStatsCalculator(Metadata metadata)
 
         return new ComposableStatsCalculator(rules.build());
     }
+
+    @Provides
+    @Singleton
+    @ForReplicatedJoin
+    public static StatsCalculator createNewReplicatedJoinStatsCalculator(Metadata metadata)
```

Comment:
Rename `createNewStatsCalculator` to `createOptimisticsStatsCalculator`. Annotate it with `@Optimistic` annotation.

Rename `createNewReplicatedJoinStatsCalculator` to `createPessimisticStatsCalculator`. Annotate it with `@Pessimistic` annotation.

Reply:
renamed `createNewStatsCalculator` to `createNewProbabilisticStatsCalculator`
and `createNewReplicatedJoinStatsCalculator` to `createNewUpperEstimateStatsCalculator`